[Android] ToolbarItems Tooltip text color

kubaflo opened the pull request:
### Description of Change

The tooltip text color is the same as the `BarTextColor`. I think it hasn't been done on purpose, so here's the proposed fix. However, if this is how it is supposed to be then, I believe, TabBarItems should set their tooltip text color based on the `TabBarTitleColor`

Thanks to relying on the native Android colors, the toolkit text will look, and adjust well to the current app theme

### Issues Fixed

Fixes https://github.com/dotnet/maui/issues/28421

|Before|After|
|--|--|
|<img src="https://github.com/user-attachments/assets/555e8f85-01f9-49ac-9a20-fd85334f39cb" width="300px"/>|<img src="https://github.com/user-attachments/assets/37e986c2-12ec-49bb-879f-185ea31557fb" width="300px"/>|

```c#
return new Window(new NavigationPage(new MainPage())
{
	BarTextColor = Colors.Red,
	Title = "Main Page"
});
```

kubaflo committed:
[Android] ToolbarItems Tooltip text color

rmarinho commented:
/azp run

jsuarezruiz commented on src/Controls/src/Core/Platform/Android/Extensions/ToolbarExtensions.cs:
```diff
@@ -375,6 +375,9 @@ internal static void UpdateMenuItemIcon(this IMauiContext mauiContext, IMenuItem
 						menuItem.SetIcon(iconDrawable);
+
+						// Setting the title, so that tooltip text is displayed in native color
+						menuItem.SetTitle(toolBarItem?.Text ?? string.Empty);
```
The change looks good, but I'm going to talk to the team to find out if we want this behavior change or delegate it to the system color.

jsuarezruiz replied:
Maybe, include it in a **Platform Specific** to allow to choose between the two different behaviors, could be a good idea.

mattleibow replied:
I think if just testting the text is changing the color, then this may be a bug or feature in android...

mattleibow commented on src/Controls/src/Core/Platform/Android/Extensions/ToolbarExtensions.cs:
```diff
@@ -375,6 +375,9 @@ internal static void UpdateMenuItemIcon(this IMauiContext mauiContext, IMenuItem
 						menuItem.SetIcon(iconDrawable);
+
+						// Setting the title, so that tooltip text is displayed in native color
+						menuItem.SetTitle(toolBarItem?.Text ?? string.Empty);
```
Don't we already set this in UpdateMenuItem?

Is there some code path that is removing it? Also, how does re-setting the text change the color?

mattleibow reviewed (changes requested):
Marking this as needs changes for now as the system needs a green or red choice to be happy. But just a 2 questions.

kubaflo committed:
Add test for ToolbarItem tooltip theme color on Android

Introduces a new test case (Issue28421) to verify that ToolbarItem tooltips use native Android theme colors instead of inheriting BarTextColor. Includes a sample page to reproduce the issue and a UI test with a reference screenshot for validation.

kubaflo commented:
<!-- AI Summary -->

## 🤖 AI Summary

<!-- SECTION:WRITE-TESTS -->
### 🧪 Test Writing

<details>
<summary><b>Attempt 1: Verifies that ToolbarItem tooltip uses native Android theme colors instead of inheriting BarTextColor. The test long-presses the toolbar item to trigger the tooltip and captures a screenshot for visual verification. ✅ Verified</b></summary>


### Test Details

| Property | Value |
|----------|-------|
| **Test Method** | `ToolbarItemTooltipUsesNativeThemeColors` |
| **Category** | `UITestCategories.ToolbarItem` |
| **Platforms** | All |
| **Status** | ✅ Verified |

### Files Created

<details>
<summary>📄 HostApp Test Page - Click to expand code</summary>

**File:** `src/Controls/tests/TestCases.HostApp/Issues/Issue28421.cs`

```csharp
File not found: src/Controls/tests/TestCases.HostApp/Issues/Issue28421.cs
```

</details>

<details>
<summary>🧪 NUnit Test - Click to expand code</summary>

**File:** `src/Controls/tests/TestCases.Shared.Tests/Tests/Issues/Issue28421.cs`

```csharp
File not found: src/Controls/tests/TestCases.Shared.Tests/Tests/Issues/Issue28421.cs
```

</details>

### Notes

## Test Details

- **Platform**: Android only (`#if ANDROID`)
- **Approach**: Uses `TouchAndHoldCoordinates()` to trigger tooltip on toolbar item, then `VerifyScreenshot()` to capture and compare
- **Baseline**: Screenshot shows tooltip "Action" with native Android theme colors (white text on dark gray background)
- **Bug reproduction**: Before the fix, tooltip text inherited `BarTextColor` (red); after the fix, tooltip uses native Android colors

## Files Created

1. **HostApp**: `Issue28421.cs` - NavigationPage with ToolbarItem and `BarTextColor = Colors.Red`
2. **NUnit Test**: `Issue28421.cs` - Screenshot test that captures tooltip appearance
3. **Snapshot**: `ToolbarItemTooltipUsesNativeThemeColors.png` - Baseline screenshot with visible tooltip

</details>
<!-- /SECTION:WRITE-TESTS -->

kubaflo committed:
Update ToolbarItemTooltipUsesNativeThemeColors.png
